eng/pipelines/templates/steps/set-cli-version-cd.yml

```diff
@@ -1,8 +1,15 @@
 steps:
-  # TODO: Handle semver prerelease and GA packages
```

Comment:
Let's keep the TODO in here around GA support. 

Use updated link: https://github.com/Azure/azure-dev/issues/344

Reply:
Based on the discussed format, I believe this script does handle GA correctly. We might just tweak tag values if needed based on #344. I don't think the TODO is necessary given the issue being tracked. I left it for now though.

```pwsh
PS D:\repos\azure-dev> Get-Content .\cli\version.txt
0.1.0-beta.6
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "BatchedCI" -BuildId 123 && git co -- .\cli\version.txt
Set version.txt contents to: 0.1.0-beta.6-daily.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "PullRequest" -BuildId 123 && git co -- .\cli\version.txt
Set version.txt contents to: 0.1.0-beta.6-pr.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "Manual" -BuildId 123 && git co -- .\cli\version.txt     
Skipping prerelease tagging for release build.

PS D:\repos\azure-dev> Get-Content .\cli\version.txt
1.0.0
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "BatchedCI" -BuildId 123 && git co -- .\cli\version.txt  
Set version.txt contents to: 1.0.0-daily.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "PullRequest" -BuildId 123 && git co -- .\cli\version.txt
Set version.txt contents to: 1.0.0-pr.123
PS D:\repos\azure-dev> eng\scripts\Set-CliVersionInBuild.ps1 -BuildReason "Manual" -BuildId 123 && git co -- .\cli\version.txt     
Skipping prerelease tagging for release build.
```